libcxx/include/__algorithm/comp.h

```diff
@@ -26,8 +27,8 @@ struct __equal_to {
   }
 };
 
-template <class _Lhs, class _Rhs>
-struct __is_trivial_equality_predicate<__equal_to, _Lhs, _Rhs> : true_type {};
+template <>
+struct __desugars_to<__equal_to, std::equal_to<>> : true_type {};
```

Comment:
One consideration is that since `equal_to<>` isn't valid in all standard modes, we should probably explicitly use `equal_to<void>` explicitly in places that need to work in C++03.